backend/controller/door.go

```diff
@@ -0,0 +1,38 @@
+package controller
+
+import (
+	"api/helpers"
+	"api/models"
+	"encoding/json"
+	"github.com/go-chi/chi/v5"
+	"net/http"
+)
+
+func GetAllDoors(w http.ResponseWriter, _ *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	err, doors := helpers.GetAllTable(w, []models.Door{})
+	if err == nil {
+		helpers.JsonWriter(w, doors)
+	}
```

Comment:
Shouldn't we do something if err != nil?

Reply:
I see you use this pattern in a lot of places. I think writing a function to handle this (and also handle an error) would be a good idea.

Reply:
yes because, the getAlltables function returns if it had an error, if it had an error it will handle it but then obv u dont wna call the writer if there was an error (works the same as return on error). I think a function is redundant in that case for these reasons

Reply:
This  is the case for most of the helper functions, in general the idea is they'll handle all their own errors and then return if they errored like in basic what the error was, so all this fn need to do is go "oh shit it errored dont continue"

Reply:
Bit of a nitpick, but it's more idiomatic to handle the error with `err != nil` and `return` early if necessary. Not a big deal though.